src/Avalonia.Input/PenDevice.cs

```diff
@@ -0,0 +1,414 @@
+            if (_position == null )
+            {
+                if (root.PointerOverElement != null)
+                    ClearPointerOver(this, 0, root, PointerPointProperties.None, KeyModifiers.None);
+                return;
+            }
+            
+            
+            var clientPoint = root.PointToClient(_position.Value);
+
+            if (rect.Contains(clientPoint))
+            {
+                if (_pointer.Captured == null)
+                {
+                    SetPointerOver(this, 0 /* TODO: proper timestamp */, root, clientPoint,
+                        PointerPointProperties.None, KeyModifiers.None);
+                }
+                else
+                {
+                    SetPointerOver(this, 0 /* TODO: proper timestamp */, root, _pointer.Captured,
+                        PointerPointProperties.None, KeyModifiers.None);
+                }
+            }
+        }
+        
+        private void ProcessRawEvent(RawPointerEventArgs e)
+        {
+            e = e ?? throw new ArgumentNullException(nameof(e));
+
+            var pen = (PenDevice)e.Device;
+            if(pen._disposed)
+                return;
+
+            _position = e.Root.PointToScreen(e.Position);
+            var props = CreateProperties(e);
+            var keyModifiers = KeyModifiersUtils.ConvertToKey(e.InputModifiers);
+            switch (e.Type)
+            {
+                case RawPointerEventType.LeaveWindow:
+                    LeaveWindow(pen, e.Timestamp, e.Root, props, keyModifiers);
+                    break;
+                case RawPointerEventType.LeftButtonDown:
+                    e.Handled = PenDown(pen, e.Timestamp, e.Root, e.Position, props, keyModifiers);
+                    break;
+                case RawPointerEventType.LeftButtonUp:
+                    e.Handled = PenUp(pen, e.Timestamp, e.Root, e.Position, props, keyModifiers);
+                    break;
+                case RawPointerEventType.Move:
+                    e.Handled = PenMove(pen, e.Timestamp, e.Root, e.Position, props, keyModifiers);
+                    break;
+            }
+        }
+
+        private void LeaveWindow(IPenDevice device, ulong timestamp, IInputRoot root, PointerPointProperties properties,
+            KeyModifiers inputModifiers)
+        {
+            device = device ?? throw new ArgumentNullException(nameof(device));
+            root = root ?? throw new ArgumentNullException(nameof(root));
+
+            _position = null;
+            ClearPointerOver(this, timestamp, root, properties, inputModifiers);
+        }
+
+
+        private PointerPointProperties CreateProperties(RawPointerEventArgs args)
+        {
+            var kind = args.Type switch
+            {
+                RawPointerEventType.LeftButtonDown => PointerUpdateKind.LeftButtonPressed,
+                RawPointerEventType.LeftButtonUp => PointerUpdateKind.LeftButtonReleased,
+                _ => PointerUpdateKind.Other,
```

Comment:
GetEventType returns more possible values than handled there
https://github.com/AvaloniaUI/Avalonia/pull/7412/files

I think it's technically possible to map pen buttons for these events?